lib/AST/ASTMangler.cpp

```diff
@@ -412,8 +412,7 @@ std::string ASTMangler::mangleDeclType(const ValueDecl *decl) {
 
 #ifdef USE_NEW_MANGLING_FOR_OBJC_RUNTIME_NAMES
 static bool isPrivate(const NominalTypeDecl *Nominal) {
-  return Nominal->hasAccess() &&
-         Nominal->getFormalAccess() <= AccessLevel::FilePrivate;
```

Comment:
This is not a good change; we need to be able to mangle invalid code.

Reply:
Note that this code is permanently #ifdef'd out.

Reply:
…so?